### PR TITLE
User-service logic was finished

### LINKED ISSUE #3: Add checking for unique relationships

### LINKED ISSUE #4: Add possibility to unfollow or delete the follower

### LINKED ISSUE #5: Add methods to check if you are the following/follower

### LINKED ISSUE #6: Finish user-service logic

### LINKED ISSUE #7: Add ability to count followers and followings

### PR DESCRIPTION
**Pull request by vpetrovaa**
## PR-Codex overview
This PR adds new methods to the UserService and creates two new interfaces, FollowerService and FollowingService, which extend UserService. It also adds new queries to the UserRepository and new mappings to the UserController.

### Detailed summary
- New methods added to the UserService interface
- Creation of FollowerService and FollowingService interfaces that extend UserService
- New queries added to the UserRepository
- New mappings added to the UserController

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/main/java/com/solvd/twitter/service/UserService.java [hunk 1]
```diff
@@ -6,6 +6,14 @@
 
 public interface UserService {
```
Comment (h1alexbel) [inline src/main/java/com/solvd/twitter/service/UserService.java:5]: @vpetrovaa try to design your services with less method amount

File: src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -18,6 +18,26 @@ public class UserServiceImpl implements UserService {
...
+    public Boolean isAFollower(final String id, final String followerId) {
+        return userRepository.isAFollower(id, followerId);
+    }
+
+    @Override
+    public Boolean isAFollowing(final String id, final String followingId) {
+        return userRepository.isAFollowing(id, followingId);
+    }
+
+    @Override
```
Comment (h1alexbel) [inline src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:36]: @vpetrovaa does neo4j support some sort of transactions?

File: src/main/java/com/solvd/twitter/service/UserService.java (hunk 1 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/twitter/service/UserService.java:5]: @h1alexbel you mean to divide it into some interfaces?

File: src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:31]: @vpetrovaa, yes, you can check if the relationship exists and return boolean result with the help of query

Comment (h1alexbel) [reply src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:36]: @vpetrovaa I read, that neo4j supports ACID to the full extent

can we use this transactional logic here, like in Spring Data + PostgreSQL via `@Transactional` annotation

Comment (vpetrovaa) [reply src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:36]: @h1alexbel is this annotation useful here?

Comment (h1alexbel) [reply src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:36]: @vpetrovaa It enabling database transactions

but does it work with neo4j?

Comment (vpetrovaa) [reply src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:36]: @h1alexbel so I checked, everything works

File: src/main/java/com/solvd/twitter/service/UserService.java (hunk 1 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/twitter/service/UserService.java:5]: @h1alexbel https://github.com/vpetrovaa/twitter/pull/8/commits/20f41dfb51363c9421813995011006a4dd2b71df

File: src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (h1alexbel) [reply src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:36]: @vpetrovaa why did you remove the annotations then?

Comment (vpetrovaa) [reply src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:36]: @h1alexbel I don't understand why we need it here. This annotation means that one method is one transaction, so if a part fails, everything done before will be rollback, isn't it? And if this query to return you are the follower or not fails, we have nothing to rollback

Comment (h1alexbel) [reply src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:36]: @vpetrovaa yes, but Transactional annotation has `readOnly` parameter

Comment (vpetrovaa) [reply src/main/java/com/solvd/twitter/service/impl/UserServiceImpl.java:36]: @h1alexbel, ok, I understood the purpose of this annotation https://github.com/vpetrovaa/twitter/pull/8/commits/895c17dea9ae33411a6d03b10cba897007def0cb

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**